### DISCUSSION TITLE
Allow HA to control text style, to help with dark mode

### DISCUSSION THREAD

**Discussion by Kallb123** [ExperienceLovelace/ha-floorplan#215] [Q&A] [answered]
I've managed to get a floor plan running, which is fantastic and helpful. Some of my sensors are located over a transparent background. This means when home assistant goes into dark mode, I can't see black text. Is there a class or method I can apply to the text fields that mean they can change colour to match HA text?

**Comment by exetico** [comment, marked as answer]
Please start by sharing your current text solution. Else, its a bit hard to help you out.

It's for sure possible to get the theme-state or so, bug I dont really think thats the best solution.

In my own solution the temperature color are shown on basis of the theme, if I remember correctly.

**Reply by Kallb123** [reply]
You can basically imagine "my first SVG floor plan" as I only started looking at this the other day. So my "text solution" is just the text object added in Inkscape. It just has a black colour. The value is driven by things like temperature entities within the HA card.

Would you mind providing any more detail on the "theme state" you mentioned and whether this can drive text colours?

I'm guessing I could check the theme and then use the Set Class function to update the text style?

**Reply by Kallb123** [reply]
I don't think I can actually detect the theme, since it's one theme with a dark and light mode?

Is there a particular class or style I can apply to the SVG text that makes it follow the rest of the HA text style?

**Reply by OzGav** [reply]
Also do you need the text over a transparent background?

**Reply by Kallb123** [reply]
I would prefer to keep it transparent because otherwise I'll be stuck with a bright white background burning my eyes in dark mode? Unless there's a way to have the background follow the light/dark mode and then the text be the opposite?

**Reply by OzGav** [reply]
I have looked and I don't see an obvious way to determine what mode HA is in but maybe @exetico knows? If you are using AUTO then that follows the device mode so if you know what logic the device uses you could set a rule to follow that and adjust the text colour accordingly (of course if you are using more than one device this may fail if they use different logic). Alternatively, take control of the theme setting within HA https://www.home-assistant.io/integrations/frontend/ and keep your floorplan in sync with that.

**Reply by Kallb123** [reply]
I am using multiple devices. My desktop never seems to be in dark mode, but my mobile is always in dark mode. I'm really surprised there's not just a dark mode css class that I can apply, or a class I can apply that makes it render in the same colour as home assistant text.

**Reply by OzGav** [reply]
Applying the CSS is easy. The issue I think you have it working out what mode your device is in. If you are happy to always be in dark mode then you adjust that by clicking on the circle in the bottom left of the screen and change it in your profile.

**Reply by exetico** [reply]
It's actually not that easy, if I understand your request. Each user has the option to change theme and mode - therefore it's not an option to fetch directly from HA.

_But it sounds like we can find a work-around for your need._:

<hr>
One option is to use `browser_mod` and fetch the `darkMode: true` attribute for the current devices sensor in Home Assistant.

Another and more simple solution is to just listen to the browsers informations.

In Javascript you're able to spot darkMode with something like:
```js
if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
 console.log("Yes, darkMode");
}
```

But that's actually totally overkill if you're already considering going down that road. Cause it's a baked in [CSS media feature](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme).

Take a look at the link. It's fairly easy to work with, and here you could just add `.whateverClassYouWouldLike` to the text-object, and style it like the following:

```css
.whateverClassYouWouldLike   { color: "lightgrey"}
@media (prefers-color-scheme: dark) {
  .whateverClassYouWouldLike{color:white;}
}

@media (prefers-color-scheme: light) {
  .whateverClassYouWouldLike{color:black;}
}
```

There's a good suggestion on how to use `.day` and `.night` in the link, too.

**Reply by Kallb123** [reply]
Ah thank you, that CSS media query looks exactly like what I need! I'm familiar with CSS, but didn't know about the media query. If that works through all my devices then that will be amazing. I'll give that a try soon and let you know. Thanks again.

**Reply by Kallb123** [reply]
Thanks @exetico. You cracked it. Those media queries combined with a certain class on the text objects made them completely flexible. It works across mobile and desktop, including the HA app :)

I just had to use fill and stroke instead of color.

**Reply by exetico** [reply]
Thats great to hear 😊!
